qcfractal/storage_sockets/mongoengine_socket.py

```diff
@@ -457,14 +457,15 @@ def get_molecules(self, molecule_ids, index="id"):
             molecule_ids = [molecule_ids]
 
         bad_ids = []
-        if index == "_id":
+        if index == "id":
             molecule_ids, bad_ids = _str_to_indices_with_errors(molecule_ids)
 
         # Project out the duplicates we use for top level keys
         proj = {"molecule_hash": False, "molecular_formula": False}
 
         # Make the query
-        data = self._tables["molecules"].find({index: {"$in": molecule_ids}}, projection=proj)
+        query = {index+'__in': molecule_ids}
+        data = Molecule.objects(**query).exclude(*proj).as_pymongo()
```

Comment:
Shouldn't this be `only` instead of `exclude`?

Reply:
Yes, thanks.

Reply:
I checked back and the original method was removing the projection, so it's `exclude` not `only`.

Reply:
Ah, I see. Can we change `proj` to a list then so we are not unpacking keys in a dict?